lua/nvim-treesitter/shell_command_selectors.lua

```diff
@@ -279,7 +279,11 @@ end
 
 function M.make_directory_change_for_command(dir, command)
   if fn.has "win32" == 1 then
-    return string.format("pushd %s & %s & popd", cmdpath(dir), command)
+    if vim.o.shell == "cmd.exe" then
```

Comment:
is this a reliable way to detect that we're executing in powershell? Could shell be `C:\Windows\system32\cmd.exe`?

Reply:
I think it would be simpler to just execute every command in `iter_cmd_sync` either in `cmd.exe` or `powershell.exe`

Reply:
Similar to https://github.com/nvim-treesitter/nvim-treesitter/blob/598b878a2b44e7e7a981acc5fc99bf8273cd1d0f/lua/nvim-treesitter/shell_command_selectors.lua#L20-L22

Reply:
> is this a reliable way to detect that we're executing in powershell? Could shell be `C:\Windows\system32\cmd.exe` or just `cmd`?

I was wondering the same thing. In `:h shell` it says, that `cmd.exe` is default on Windows, but I guess one could specify a full path which would mess this solution up.

Reply:
> I think it would be simpler to just execute every command in `iter_cmd_sync` either in `cmd.exe` or `powershell.exe`

I'll try that out!

Reply:
maybe you could also regex on the shell like `cmd(.exe)?$`

Reply:
I guess, even using regex might lead to open edge cases. I just implemented your suggestion of using `cmd /C` :)